flake.nix

```
@@ -53,7 +52,10 @@
       # attribute it displays `omitted` instead of evaluating all packages,
       # which keeps `nix flake show` on Nixpkgs reasonably fast, though less
       # information rich.
-      legacyPackages = forAllSystems (system: import ./. { inherit system; });
+      legacyPackages = forAllSystems (system: import ./. {
+        inherit system;
+        overlays = [ self.overlays.setLibVersionInfo ];
+      });
```

I believe this is not entirely backwards compatible, because if you use `--impure`, `overlays` defaults to stuff from `~/.config/nixpkgs`, whereas passing the argument directly will prevent that from happening. So instead we should just extend it with the overlay:
```suggestion
      legacyPackages = forAllSystems (system:
        (import ./. { inherit system; }).extend self.overlays.setLibVersionInfo
      );
```

---

Nice catch, thanks!